### PR TITLE
wget2 2.1.0

### PR DESCRIPTION
**Pull request by equal-l2**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/w/wget2.rb [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-  url "https://gitlab.com/gnuwget/wget2/uploads/83752270de83e103306576e67a1c7c80/wget2-2.0.1.tar.gz"
-  sha256 "0bb7fa03697bb5b8d05e1b5e15b863440826eb845874c4ffb5e32330f9845db1"
+  url "https://ftp.gnu.org/gnu/wget/wget2-2.1.0.tar.gz"
+  sha256 "a05dc5191c6bad9313fd6db2777a78f5527ba4774f665d5d69f5a7461b49e2e7"
   license "GPL-3.0-or-later"
 
+  livecheck do
+    url "https://ftp.gnu.org/gnu/wget/"
+    strategy :gnu
+    regex(%r{href=.*?wget2-(\d+(?:\.\d+)*)(?:\.[a-z]+|/)}i)
```
Comment (samford) [inline Formula/w/wget2.rb:11]:
```suggestion
    regex(/href=.*?wget2[._-]v?(\d+(?:\.\d+)+)\.t/i)
```

We only use the `#strategy` method when A) we need to use a strategy other than the one that livecheck uses for the URL or B) we need to use a `strategy` block. In this case, livecheck already uses the `Gnu` strategy for this URL, so this `#strategy` call is unnecessary.

I've added a modified regex above that aligns with our typical regexes for matching a file from an `href` attribute on an HTML page. Namely:

* We use `[._-]` as the delimiter between the name (`wget2`) and the version.
* We use `v?(\d+(?:\.\d+)+)` as the standard regex for versions like `1.2.3`/`v1.2.3`. We only use the looser `*` format when we specifically need to match a version that is usually like `1.2` but can also be `1` (this is rare).
* We typically use a loose `\.t` as the trailing anchor for tarballs.

**Review by samford — CHANGES_REQUESTED** [review]
Thanks for this contribution! I had been planning to open a PR for this myself, so I've added a suggestion with tweaks to the `livecheck` block to bring it in line.

Once the requested changes are integrated, please squash this down to one "wget2 2.1.0" commit.

File: Formula/w/wget2.rb (hunk 1 above)
Comment (equal-l2) [reply Formula/w/wget2.rb:11]: Thank you! I'll adopt your suggestion.

The regex I use is from Homebrew (the one used in Gnu strategy) with some simplification, so I am sure it will be beneficial to apply your change to brew too.

Comment (samford) [reply Formula/w/wget2.rb]: > The regex I use is from Homebrew (the one used in Gnu strategy) with some simplification, so I am sure it will be beneficial to apply your change to brew too.

The `Gnu` strategy's default regex has to be somewhat loose in places to match a wider variety of potential scenarios. In a `livecheck` block, we cater the regex to the context of the source we're checking, so it's more specific than the default.

We already use `[._-]` as the aforementioned delimiter in the `Gnu` strategy and we can't make guarantees around the version format or file extension, so the existing strategy regex is still probably fine (unless/until we identify areas for improvement across all formulae using the `Gnu` strategy). I appreciate the suggestion, though.

File: Formula/w/wget2.rb [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
   desc "Successor of GNU Wget, a file and recursive website downloader"
   homepage "https://gitlab.com/gnuwget/wget2"
-  url "https://gitlab.com/gnuwget/wget2/uploads/83752270de83e103306576e67a1c7c80/wget2-2.0.1.tar.gz"
-  sha256 "0bb7fa03697bb5b8d05e1b5e15b863440826eb845874c4ffb5e32330f9845db1"
+  url "https://ftp.gnu.org/gnu/wget/wget2-2.1.0.tar.gz"
+  sha256 "a05dc5191c6bad9313fd6db2777a78f5527ba4774f665d5d69f5a7461b49e2e7"
   license "GPL-3.0-or-later"
 
+  livecheck do
+    url "https://ftp.gnu.org/gnu/wget/"
```
Comment (samford) [inline Formula/w/wget2.rb:9]:
```suggestion
    url :stable
```

Sorry for the additional review but I overlooked this previously. By default, livecheck will use the `Gnu` strategy with the `stable` URL and check the same URL that's used in this `livecheck` block, so we simply use `url :stable` here.

Comment (equal-l2) [reply Formula/w/wget2.rb:9]: I'm somewhat hesitant to do that because the stable URL scheme changed just from 2.1.0.
Before the version the stable URL points to GitLab upload, and I'm not sure if the scheme change is permanent or not. 
The duplicate URL is a future-proof for when the URL won't point to the GNU page in the later version.

Comment (samford) [reply Formula/w/wget2.rb:9]: I understand your concern but if the `stable` URL changes to a different source in a subsequent PR, then the `livecheck` block would need to be updated in the same PR [to align with the new `stable` source] regardless. Using `url :stable` is more likely to cause livecheck to return an `Unable to get versions` error if the `stable` URL changes to a different source and this error will be surfaced from our CI. The error would need to be addressed before the PR is merged, so it helps to ensure the `livecheck` block is kept in line with the `stable` source.

Using a string literal that duplicates the URL from the `Gnu` strategy _would_ allow the check to continue to work if the `stable` URL changes to a different source but that's not actually desirable for us (as counterintuitive as it may seem). In that scenario, a PR changing the `stable` source may pass CI and end up being merged despite the `livecheck` block not checking the `stable` source. This type of issue is harder to identify than a check that's visibly failing.

Long story short, we should use `url :stable` here and if the `stable` source changes again in the future, we will simply update (or remove) the `livecheck` block at the same time. This is standard procedure for `livecheck` blocks in both homebrew/core and homebrew/cask and this one isn't an exception.

Does this make sense?

Comment (equal-l2) [reply Formula/w/wget2.rb:9]: It perfectly makes sense, thank you for the detailed explanation!

> The error would need to be addressed before the PR is merged

I didn't know this.
If actions can catch the error, I'm fine with the change.

Comment (equal-l2) [reply Formula/w/wget2.rb]: The change is applied and squashed.

**Review by samford — APPROVED** [review]
Thanks, @equal-l2!
